tests/unit/expiry/TestDataProvider.unitTest.ts

```diff
@@ -49,17 +56,87 @@ describe("TestDataProvider", () => {
           testDataProvider.testResultsDAO = new MockTestResultsDAO();
           await testDataProvider.getTestHistory("123");
         } catch (error) {
-          expect.assertions(1);
           expect(error).toEqual(
-            new HTTPError(500, MESSAGES.INTERNAL_SERVER_ERROR)
+            new HTTPError(500, enums.MESSAGES.INTERNAL_SERVER_ERROR)
           );
         }
       });
     });
+    describe("when getBySystemNumber returns invalid testEndTimestamp", () => {
+      it("should throw error", async () => {
+        expect.assertions(1);
+        try {
+          testDataProvider = new TestDataProvider();
+          MockTestResultsDAO = jest.fn().mockImplementation(() => {
+            return {
+              getBySystemNumber: () => {
+                return {
+                  Count: 1,
+                  Items: [
+                    {
+                      testStartTimestamp: "2019-03",
+                      testEndTimestamp: "2019-03-10T08:47:59.269Z",
+                      testTypes: [
+                        { testCode: "fft1", testExpiryDate: "2019-02-10T08:47:59.261009Z" }
+                      ],
+                      testStatus: enums.TEST_STATUS.SUBMITTED
+                    }
+                  ]
+                };
+              }
+            };
+          });
+          testDataProvider.testResultsDAO = new MockTestResultsDAO();
+          const result = await testDataProvider.getTestHistory("123");
+          console.log(result);
+        } catch (error) {
+          expect(error).toEqual(
+            new HTTPError(500, enums.MESSAGES.INTERNAL_SERVER_ERROR)
+          );
+        }
+      });
+    });
+    describe("when getBySystemNumber returns array of valid data", () => {
```

Comment:
do we want to add one more when we have 000000 too so we cover our 3 patterns?

Reply:
rephrase the description with format and add scenario with 3 digits.